vendor/bat-native-ads/src/bat/ads/internal/account/account.cc

```diff
@@ -95,24 +96,26 @@ void Account::MaybeGetIssuers() const {
   issuers_->MaybeFetch();
 }
 
-void Account::DepositFunds(const std::string& creative_instance_id,
-                           const AdType& ad_type,
-                           const ConfirmationType& confirmation_type) {
+void Account::Deposit(const std::string& creative_instance_id,
+                      const AdType& ad_type,
+                      const ConfirmationType& confirmation_type) {
   DCHECK(!creative_instance_id.empty());
   DCHECK_NE(AdType::kUndefined, ad_type.value());
   DCHECK_NE(ConfirmationType::kUndefined, confirmation_type.value());
 
-  database::table::CreativeAds database_table;
-  database_table.GetForCreativeInstanceId(
-      creative_instance_id,
-      [=](const bool success, const std::string& creative_instance_id,
-          const CreativeAdInfo& creative_ad) {
+  std::unique_ptr<DepositBase> deposit =
+      DepositsFactory::Build(confirmation_type);
+  DCHECK(deposit);
+
+  deposit->GetValue(
```

Comment:
The issue was that previously we were fetching the creative value from the creative ads database table, we should have only fetched the value if this was a cash deposit. Views are treated as cash deposits, i.e. value is > 0.0. Other confirmation types should have been treated as non-cash deposits, which should not be looked up in the database.

Reply:
cc @btlechowski